Update Load.php

Underdog-01 opened the pull request:
SQL column does not appear to have a default value of '' and/or NOT NULL resulting in a possible undefined log error.
I came across this issue on a test forum while accessing part of the admin section.

Signed-off-by: Underdog-01 <github.underdog@gmail.com>

Underdog-01 committed:
Update Load.php

SQL column does not appear to have a default value of '' and/or NOT NULL resulting in a possible undefined log error.
I cam across this issue on a test forum while accessing part of the admin section.

Signed-off-by: Underdog-01 <github.underdog@gmail.com>

Underdog-01 committed:
fix end of file

Signed-off-by: Underdog-01 <github.underdog@gmail.com>

Sesquipedalian commented on Sources/Load.php:
```diff
@@ -1280,10 +1280,10 @@ function loadMemberData($users, $is_name = false, $set = 'normal')
 
 			// Take care of proxying avatar if required, do this here for maximum reach
-			if ($image_proxy_enabled && !empty($row['avatar']) && stripos($row['avatar'], 'http://') !== false)
+			if ((!empty($row['avatar'])) && $image_proxy_enabled && stripos($row['avatar'], 'http://') !== false)
```
What's the point of this change?

Underdog-01 replied:
To ensure the column has a value prior to using stripos .. I did not see an error from that line but put the edit in to cover the possibility as I have seen undefined errors with my own projects on similar logic. The brackets could go around the first 2 variables or your other suggestion I also thought of prior can be implemented which will cover any null possibility.

Sesquipedalian replied:
As @jdarwood007 said below,

>There isn't a case where that can happen though as you just flipped the first 2 checks around and the strpos is the 3rd check.

This change doesn't actually do anything.

Sesquipedalian commented:
If the avatar column can be NULL, then we probably want to change the installer and upgrader files to make it a NOT NULL.

Underdog-01 commented:
I can implement:  ~ not null default '' ~ but prior to doing that I need to check/test that there is not an attempt to insert a null value anywhere in the current SMF 2.1 coding. I will test things out this weekend & implement the edits unless someone else reads this & beats me to it.

jdarwood007 commented:
There isn't a case where that can happen though as you just flipped the first 2 checks around and the strpos is the 3rd check.

Underdog-01 committed:
2nd change reverted

short-circuit operator stops prior to evaluating subsequent conditions
when $row['avatar'] is empty

Underdog-01 committed:
edit commit message

Signed-off-by: Underdog-01 <github.underdog@gmail.com>

XinYenFon commented:
extra txt.

Sesquipedalian commented:
This branch is kind of messed up now, @Underdog-01. If you don't mind, please make a fresh branch, reimplement your finalized changes in that, and then submit it as a new PR.

Underdog-01 commented:
Agreed.. but it will have to be Friday evening.

Sesquipedalian commented:
No problem. Thanks.
